File: app/src/main/kotlin/com/deepseek/harness/android/ui/models/ModelsUiState.kt
package com.deepseek.harness.android.ui.models

import androidx.compose.runtime.Immutable
import com.deepseek.harness.android.domain.model.ModelSelection
import com.deepseek.harness.android.domain.model.SessionModels
import com.deepseek.harness.android.domain.model.SessionSummary

@Immutable
data class ModelsUiState(
    val sessions: List<SessionSummary> = emptyList(),
    val selectedSessionId: String? = null,
    val models: SessionModels? = null,
    val selected: ModelSelection? = null,
    val isLoading: Boolean = false,
    val errorMessage: String? = null,
)

sealed interface ModelsAction {
    data class SelectSession(val sessionId: String) : ModelsAction
    data class SelectModel(val provider: String, val model: String) : ModelsAction
    data object Refresh : ModelsAction
    data object DismissError : ModelsAction
}

File: flutter/app/lib/ui/models/models_ui_state.dart
/// Models screen UI state and intents — port of ModelsUiState.kt.
library;

import 'package:domain/model/model_catalog.dart';
import 'package:domain/model/session.dart';

final class ModelsUiState {
  const ModelsUiState({
    this.sessions = const <SessionSummary>[],
    this.selectedSessionId,
    this.models,
    this.selected,
    this.isLoading = false,
    this.errorMessage,
  });

  final List<SessionSummary> sessions;
  final String? selectedSessionId;
  final SessionModels? models;
  final ModelSelection? selected;
  final bool isLoading;
  final String? errorMessage;
}

sealed class ModelsAction {
  const ModelsAction();
}

final class SelectModelsSession extends ModelsAction {
  const SelectModelsSession(this.sessionId);

  final String sessionId;

  @override
  bool operator ==(Object other) =>
      other is SelectModelsSession && other.sessionId == sessionId;

  @override
  int get hashCode => sessionId.hashCode;
}

final class SelectModelAction extends ModelsAction {
  const SelectModelAction(this.provider, this.model, [this.reasoningEffort]);

  final String provider;
  final String model;
  final String? reasoningEffort;

  @override
  bool operator ==(Object other) =>
      other is SelectModelAction &&
      other.provider == provider &&
      other.model == model &&
      other.reasoningEffort == reasoningEffort;

  @override
  int get hashCode => Object.hash(provider, model, reasoningEffort);
}

final class RefreshModelsAction extends ModelsAction {
  const RefreshModelsAction();

  @override
  bool operator ==(Object other) => other is RefreshModelsAction;

  @override
  int get hashCode => 'refresh-models'.hashCode;
}

final class DismissModelsError extends ModelsAction {
  const DismissModelsError();

  @override
  bool operator ==(Object other) => other is DismissModelsError;

  @override
  int get hashCode => 'dismiss-models-error'.hashCode;
}
